sklearn/datasets/_samples_generator.py

```diff
@@ -1631,28 +1631,30 @@ def make_sparse_spd_matrix(
     """
     random_state = check_random_state(random_state)
 
-    chol = -np.eye(dim)
-    aux = random_state.uniform(size=(dim, dim))
-    aux[aux < alpha] = 0
-    aux[aux > alpha] = smallest_coef + (
-        largest_coef - smallest_coef
-    ) * random_state.uniform(size=np.sum(aux > alpha))
-    aux = np.tril(aux, k=-1)
+    chol = -sp.eye(dim)
+    aux = sp.random(
+        m=dim,
+        n=dim,
+        density=alpha,
+        format="csr",
+        data_rvs=lambda x: random_state.uniform(
+            size=x, low=smallest_coef, high=largest_coef
+        ),
+        random_state=random_state,
+    )
+    aux = sp.tril(aux, k=-1, format="csc")
```

Comment:
Why are we choosing the format "csc" specifically?

Reply:
I wanted to avoid `coo` because it does not support slicing.

Reply:
I see. Does that mean that we can use "csr" as well?

Reply:
Yes that's true.

Reply:
Okay "csc" seems fine. Could you add a comment why we are setting the format?